### ISSUE TITLE
Water temp message error 

### ISSUE THREAD

**Issue by scadaperspective** [scadaperspective/Simulator_pi#10]
Water temp message must be in error

**Comment by scadaperspective** [comment]
Setting up linux box for testing...

Some issue with po files to sort 
 
Figuring out some network setup options and such to test use on networks is good.

Basic setup is one computer macOS is the “server” and it is servicing linux box. 

All messages being sent on network look ok Petri 

values display ok on dashboard 1.2 on my linux box.

Will look at why dashboard tactics isnt happy if i get some time while your travelling

**Comment by ghost** [comment]
Can you please give an example of a complete sentence? Dashboard 1.2 in OpenCPN and Dashboard in dashboard_tactics_pi are the same, this type of messages never go to Tactics-part. Why it does not show for you remains a mystery (I tested on Windows, BTW, using my boat's instrument data, it is a Raymarine depth sounder which sends the water temperature, it needs to go from SeaTalk to NMEA converter box. You'll find from my original ticket the recording to compare with yours. You can play that on you Linux box. Among other possibilities. Now need to start move, I'll be on-line, of course, but not being able to test.

**Comment by scadaperspective** [comment]
Ok Petri, 

This has been worthwhile for me.

 I have made some changes to the CRC routines out of looking into what may be causing the dashboard_tactic's plugin to not be reading the message relating too water and air temp. 

 Something I wanted to tidy up was that ship driver (the framework I have used to save me a lot of time) was inserting a line feed before the carriage return and linefeed you expect to have at the end of the message.....

I am now using the form as to what most plugin's seem to use for generating the CRC for the AIS message and the NMEA0183 messages. looks to be generating CRC and cr/lf with no extraneous lf before it now. 


The message type IIXDR permits stringing together of multiple values in a single message in the standard. The built in dashboard_pi reads the sentence fine. 

FWIW I have implemented this message frame format for the Pitch and Heel message and for Water and Air Temp messages.

Pitch and heel work fine on both ....

Ok re your ray marine it is sending sounder and water temp. If you know the message type i can emulate that instrument for you if you would like.
